### PR TITLE
publish: convert to send messages to Matrix

### PR DESCRIPTION
**Pull request by sumnerevans**
WIP, but this converts to post notifications to Matrix instead of IRC.

It's not really very scalable at this point to multiple rooms. The parameters as exists right now is as follows:

* `SERVER` - the matrix server to use (for example `https://nixos.org`)
* `ROOM_ID` - the room ID to send the messages to (for example `!wfRWMLdXruNtsaDaKU:sumnerevans.com`)
* `TOKEN` - the API token for the bot user

Bot testing room: https://matrix.to/#/#nixos-bot-testing:sumnerevans.com?via=sumnerevans.com&via=nixos.org

Example:
```
$ SERVER=<matrix-server> \
> ROOM_ID='<roomid>' \
> TOKEN=<insert-token-here> \
> ./src/calculate ~proj/nixpkgs ~tmp/nix-channel-monitor-data
```

### CODE REVIEW

File: src/calculate [hunk 1]
```diff
@@ -8,38 +8,22 @@ dbg() {
...
-      -H "content-type:application/json" \
-      -XPOST \
-      "$AMQPAPI"/api/exchanges/ircbot/amq.default/publish \
-      --data @-
+  msgplain=$(cat | head -n1)
+  msgformatted=$(cat | head -n1)
+
+  # Ensure that the bot is in the room.
+  curl -s -X POST \
+    "$SERVER/_matrix/client/r0/join/$ROOM_ID?access_token=$TOKEN" > /dev/null
```
Comment (sumnerevans) [inline src/calculate:17]: I don't like this. It's joining the room before sending every message. Not sure where to put it, though.

Comment (grahamc) [reply src/calculate:17]: Is there a way to see if we are part of the room first, and only join if we're not?

Comment (sumnerevans) [reply src/calculate:17]: Yes, we could use this and parse the JSON. https://matrix.org/docs/spec/client_server/r0.6.0#get-matrix-client-r0-joined-rooms

The more I work on this, the more I want to rewrite in something that isn't bash haha

Comment (grahamc) [reply src/calculate:17]: Please do! Or at least the parts of it that you need to!

**Comment by sumnerevans** [comment]
Closing in favor of a complete rewrite of `calculate` in Python.

**Pull request closed without merging**